integrationTests/vm/esdt/esdtNft_test.go

```diff
@@ -0,0 +1,703 @@
+		idxProposers[i] = i * nodesPerShard
+	}
+	idxProposers[numOfShards] = numOfShards * nodesPerShard
+
+	integrationTests.DisplayAndStartNodes(nodes)
+
+	defer func() {
+		_ = advertiser.Close()
+		for _, n := range nodes {
+			_ = n.Messenger.Close()
+		}
+	}()
+
+	initialVal := big.NewInt(10000000000)
+	integrationTests.MintAllNodes(nodes, initialVal)
+
+	round := uint64(0)
+	nonce := uint64(0)
+	round = integrationTests.IncrementAndPrintRound(round)
+	nonce++
+
+	///////////------- send token issue
+
+	issueNFT(nodes, core.NonFungibleESDT)
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard := 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	tokenIdentifier := string(getTokenIdentifier(nodes))
+
+	//// /////// ----- set special role
+	setRole(nodes, nodes[1].OwnAccount.Address, []byte(tokenIdentifier), []byte(core.ESDTRoleNFTCreate))
+	setRole(nodes, nodes[1].OwnAccount.Address, []byte(tokenIdentifier), []byte(core.ESDTRoleNFTBurn))
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard = 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	nftMetaData := nftArguments{
+		name:       []byte("nft name"),
+		quantity:   1,
+		royalties:  9000,
+		hash:       []byte("hash"),
+		attributes: []byte("attr"),
+		uri:        [][]byte{[]byte("uri")},
+	}
+	createNFT([]byte(tokenIdentifier), nodes[1], nodes, &nftMetaData)
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard = 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	checkNftData(
+		t,
+		nodes[1].OwnAccount.Address,
+		nodes[1].OwnAccount.Address,
+		nodes,
+		tokenIdentifier,
+		&nftMetaData,
+		false,
+	)
+
+	// increase quantity
+	nonceArg := hex.EncodeToString(big.NewInt(0).SetUint64(1).Bytes())
+	quantityToBurn := int64(1)
+	quantityToBurnArg := hex.EncodeToString(big.NewInt(quantityToBurn).Bytes())
+	txData := []byte(core.BuiltInFunctionESDTNFTBurn + "@" + hex.EncodeToString([]byte(tokenIdentifier)) +
+		"@" + nonceArg + "@" + quantityToBurnArg)
+	integrationTests.CreateAndSendTransaction(
+		nodes[1],
+		nodes,
+		big.NewInt(0),
+		nodes[1].OwnAccount.Address,
+		string(txData),
+		integrationTests.AdditionalGasLimit,
+	)
+
+	time.Sleep(time.Second)
+	nrRoundsToPropagateMultiShard = 5
+	nonce, round = integrationTests.WaitOperationToBeDone(t, nodes, nrRoundsToPropagateMultiShard, nonce, round, idxProposers)
+	time.Sleep(time.Second)
+
+	// the token data is removed from trie if the quantity is 0, so we should not find it
+	checkNftData(
+		t,
+		nodes[1].OwnAccount.Address,
+		nodes[1].OwnAccount.Address,
+		nodes,
+		tokenIdentifier,
+		&nftMetaData,
+		true,
+	)
+}
+
+func TestESDTSemiFungibleTokenCreateAddAndBurn(t *testing.T) {
```

Comment:
function too long

Reply:
extracted the functionality for preparing